bolt/include/bolt/Passes/PAuthGadgetScanner.h

```diff
@@ -199,19 +199,34 @@ struct Report {
   virtual void generateReport(raw_ostream &OS,
                               const BinaryContext &BC) const = 0;
 
+  virtual const ArrayRef<MCPhysReg> getAffectedRegisters() const { return {}; }
+  virtual void
+  setOverwritingInstrs(const std::vector<MCInstReference> &Instrs) {}
+
```

Comment:
My feel is that the base class `Report` shouldn't have `getAffectedRegisters` and `setOverwritingInstrs`. These seem properties that are quite specific to a "Gadget", and not necessarily relevant for all kinds of diagnostics that could be produced on a specific instruction in a binary.

This is just a thought, I'm not going to block the approval for the PR on it if there is a good reason to keep these 2 methods in the base class.

Reply:
Added a comment so far.

Reply:
Thanks for the detailed explanation. Let's keep it as is for this PR.